### PR TITLE
fix(APIv2): RHINENG-13402 enable multidimensional sort in OpenAPI spec

### PR DESCRIPTION
**Pull request by romanblanco**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: spec/swagger_helper.rb [hunk 1]
```diff
@@ -158,7 +158,7 @@ def sort_params(model = nil)
 end
 
 def sort_params_v2(model = nil, except: [])
-  parameter name: :sort_by, in: :query, required: false,
+  parameter name: :sort_by, in: :query, required: false, style: :form, explode: true,
```
Comment (skateman) [inline spec/swagger_helper.rb:161]: Why is the `style: :form` needed?

Comment (romanblanco) [reply spec/swagger_helper.rb:161]: Following rswag documentation: https://github.com/rswag/rswag/wiki/FAQ#for-openapi-3 

https://swagger.io/docs/specification/v3_0/serialization/#query-parameters

Comment (skateman) [reply spec/swagger_helper.rb:161]: Okay, still don't understand why is this needed.

Comment (romanblanco) [reply spec/swagger_helper.rb:161]:
```
form – (default) ampersand-separated values, also known as form-style query expansion. Corresponds to the {?param_name} URI template.
```

I believe this is the option we want to use

Comment (skateman) [reply spec/swagger_helper.rb:161]: If it's default, why do we need this now? The format checks out, we only need the `explode` as far as I understand.

Comment (romanblanco) [reply spec/swagger_helper.rb:161]: Technically, we should not need it, as it is supposed to be the default option.

I suggest the change to explicitly state the expansion options to rule out that the problem is occuring in the generated OpenAPI spec.

Comment (skateman) [reply spec/swagger_helper.rb:161]: And why just here be explicit? If it's a default value, it should be consistent across all parameters in the whole specification. So either change it everywhere to `style: form` or please delete this.

Comment (romanblanco) [reply spec/swagger_helper.rb]: We want this to be applied on those parameters of `type: :array`. Where else do you see this missing?

Comment (skateman) [reply spec/swagger_helper.rb:161]: If this is default (which is), it means it is applied to everything, not just the `type: :array` so I am asking again, why is this case so special that you want to have an explicit setting that is already implicitly set on it? Why and how is this one single case different from all the other parameters that may or may not have a `type: :array` which is not connected to the formatting that you are trying to set now explicitly?

Comment (romanblanco) [reply spec/swagger_helper.rb:161]: The referenced ticket answers your question.

Comment (skateman) [reply spec/swagger_helper.rb:161]: No it doesn't, you have 2 independent attributes that you are for some reason try to bind into a single context. One of them is default, so its setting is irrelevant and to maintain consistency across the OpenAPI spec, it makes no sense to explicitly specify in a single location.

**Pull request closed without merging**